### PR TITLE
Fixed various issues with Sound flags

### PR DESCRIPTION
**Pull request by Eism**
Resolves: #21974
Resolves: #21943
Resolves: #21940
Resolves: #21942

### CODE REVIEW

File: src/notation/internal/notationplayback.cpp [hunk 1]
```diff
@@ -424,8 +424,13 @@ bool NotationPlayback::doAddSoundFlag(StaffText* staffText)
     }
 
     for (EngravingObject* obj : *links) {
-        if (obj && obj->isStaffText()) {
-            toStaffText(obj)->add(soundFlag->clone());
+        if (obj && obj->isStaffText() && obj != staffText) {
+            SoundFlag* linkedSoundFlag = soundFlag->clone();
```
Comment (RomanPudashkin) [inline src/notation/internal/notationplayback.cpp:428]: As far as I can tell, you can use linkedClone() here

File: src/playback/view/internal/soundflag/soundflagsettingsmodel.cpp [hunk 2]
```diff
@@ -242,7 +244,15 @@ QVariantList SoundFlagSettingsModel::contextMenuModel()
 {
```
Comment (RomanPudashkin) [inline src/playback/view/internal/soundflag/soundflagsettingsmodel.cpp:244]: Ideally, we should not re-build the context menu model, but just update "enabled" of the reset button

**Comment by avvvvve** [comment]
Nice! #21974, #21943, and #21942 are done. I just need to move the sound flag properties section into #20183 as discussed.

#21940 (reset behavior) seems to not have changed for me.
If I have a Muse Percussion: Suspended Cymbal sound flag set to "Large - Sticks" and "Secco", "Reset to default sound" does the following:
- Changes the text to "Staff text"
- When you dismiss the popup, the sound flag is removed (so it's not actually resetting)

The expected behavior:
- The staff text changes to what we would show if the user manually selected the first option in Modify sound and the first option in Extended techniques (or just one of those sections if there is only one). i.e. for Muse Percussion: Suspended Cymbal we should see "large - mallets, ordinary" in the score
- The first Modify sound and first Extended technique options are both selected
- When you dismiss the popup, the sound flag is still there

**Comment by avvvvve** [comment]
Approved!

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**